docs/architecture.md

```diff
@@ -112,76 +112,206 @@ The redux store is composed of two `slices` : `documents` and `queries`:
+    CC->>Store: Check query
+    alt Query does not exist
+    CC->>Store: Init query
+    Store->>Store: Add query to collection
+    else Query exists
+    CC->>CC: Check fetch policy
+    CC->>CC: Check loading status
+    end
+    CC->>Store:  Load query
+    Store->>Store: Update query status
+    CC->>Backend:  Execute query
+    Backend->>CC:  Data
+    CC->>Store: Data
+    Store->>Store: Update query status
+    Store->>Store: Update documents 
+    CC->>App: Results
+    Store->>Store: Auto-update queries
+```
+
+#### Persisted store and background fetching stategy 
+
+An app can decide to persist its store in localstorage. See [here](https://github.com/cozy/cozy-home/blob/c47858515d4f95c24fa88a6b96bb0b500b947424/src/store/configureStore.js#L46) for an example.
+When doing so, the app loads the store when it starts and benefit from the instant available cached data.
+
+But, the data could have significantly evolved between the last store save and the start, making the documents potentially stale. 
+Hopefully, since the fetch policy is probably expired, queries will be run again, retrieving fresh data. Unfortunately, this might cause an immediate spinning display on the app when it starts, even though there is available data thanks to the persisted store. 
+
+To solve this, a `backgroundFetching` option can be set, at the query level, or at the store level (enabled for all queries).
+
+When the query is loaded, it simply keeps the query status in `loaded` state (rather than setting a `loading` state) and set a new `isFetching` attribute, stored in the query. 
+Thanks to this, the app can adapt its display to avoid the "spinning effect", and might inform the user that data is being updated in the background, without refreshing the whole app.
+
+See [this PR](https://github.com/cozy/cozy-client/pull/1211) for more insights.
+
+
+Here is a simplified execution sequence with or without background fetching:
+```mermaid
+---
+title: Background fetching
+---
+sequenceDiagram
+    participant App as App
+    participant CC as cozy-client
+    participant Store as Store
+    participant Backend as Backend
+
+    App->>CC: Query documents
+
+    CC->>Store:  Load query
+    alt Without background fetching
+      Store->>Store: update query status
+      Store->>App: Loading query status
+      App->>App: Spinner
+    else With background fetching
+      Store->>Store: no change on query status
+      Store->>App: isFetching status
+      App->>App: Do nothing
+    end
+    CC->>Backend:  Execute query
+    Backend->>CC:  New data
+    CC->>Store: Update store
+    CC->>App: New data
+    App->>App: Refresh display
+```
+
+### Updates
+
+Here, we detail what happens when a document is updated, i.e. a mutation happens, typically after a `client.save`.
+
+If the document `_id` does not exist in the store's `documents` collection, i.e. after the a creation, it is simply added.
+If it already exists, the document content is updated in this collection.
```

Comment:
after the roundtrip with the server? or this is an optimistic update?

Reply:
It is after the "server" (how do we call it to include pouch ? backend?) response: https://github.com/cozy/cozy-client/blob/master/packages/cozy-client/src/CozyClient.js#L1057-L1068

Reply:
Yeah, I'll call that the backend. We don't care if this is the cozy-stack or the Pouch. But beware that the returned document can be different depending on the backend (aka if the stack adds a few information for instance)

Reply:
> But beware that the returned document can be different depending on the backend

Indeed, but ideally, there should be no difference from the app perspective. In the future, I assume we might have some FileCollection in the pouch link to handle files path, for example.